experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -447,22 +481,70 @@ function parent_type(::Type{WeylGroupElem})
   return WeylGroup
 end
 
-# rename to reduced decompositions ?
+@doc raw"""
+    reduced_expressions(x::WeylGroupElem; up_to_commutation::Bool=false) -> ReducedExpressionIterator
+
+Return an iterator over all reduced expressions of `x`.
+
+If `up_to_commutation` is `true`, the iterator will not return an expression that only
+differs from a previous one by a swap of two adjacent commuting simple reflections.
+
+# Examples
+```jldoctest
+julia> W = weyl_group(:A, 3);
+
+julia> x = W([1,2,3,1]);
+
+julia> collect(reduced_expressions(x))
+3-element Vector{Vector{UInt8}}:
+ [0x01, 0x02, 0x03, 0x01]
+ [0x01, 0x02, 0x01, 0x03]
+ [0x02, 0x01, 0x02, 0x03]
+
+julia> collect(reduced_expressions(x; up_to_commutation=true))
+2-element Vector{Vector{UInt8}}:
+ [0x01, 0x02, 0x03, 0x01]
+ [0x02, 0x01, 0x02, 0x03]
+```
+The second expression of the first iterator is not contained in the second iterator
+because it only differs from the first expression by a swap of two the two commuting simple reflections `s1` and `s3`.
+"""
 function reduced_expressions(x::WeylGroupElem; up_to_commutation::Bool=false)
   return ReducedExpressionIterator(x, up_to_commutation)
 end
 
 @doc raw"""
     word(x::WeylGroupElem) -> Vector{UInt8}
+
+Return `x` as a list of indices of simple reflections, in reduced form.
+
+This function is right inverse to calling `(W::WeylGroup)(word::Vector{<:Integer})`.
 """
 function word(x::WeylGroupElem)
   return x.word
 end
 
+@doc raw"""
+    fp_group(W::WeylGroup) -> FPGroup
+
+Construct a group of type `FPGroup` that is isomorphic to `W`.
+
+This is achieved by constructing a free group with the same rank as `W`,
+and adding relations corresponding to the Coxeter relations of `W`.
```

Comment:
Is this a promise for the output or just a description what happens internally (without guarantees that this won't change)? If it is the former, I think it would be better to rephrase it as: The fp group will have blabla generators and blabla relations.

Reply:
Similar for the other method if the aim is not find any isomorphism with an fp group, but a specific one.

Reply:
```suggestion
The `FPGroup` will be the quotient of a free group with the same rank as `W`,
where we have the natural 1-to-1 correspondence of generators, modulo the Coxeter relations of `W`.
```
something like this?

Reply:
Yes, sounds good! I would know what is meant.